### ISSUE TITLE
GNOME 45 support

### ISSUE THREAD

**Issue by NastyaGrifon** [BigE/desk-changer#98]
version 45 is not backwards compatable with plugins made for older versions and requires significant adaptation.  

Right now the desk-changer extension cannot be enabled, the settings menu crashes with

```
SyntaxError: export declarations may only appear at top level of a module

Stack trace:
  @file:///usr/share/gnome-shell/extensions/desk-changer@eric.gach.gmail.com/prefs.js:4:16
  _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
```

Other  projects face similar issues: [wintile ](https://github.com/Fmstrat/wintile/issues/165), [CoverflowAltTab](https://github.com/dmo60/CoverflowAltTab/issues/208)

Useful resources: 
https://gitlab.gnome.org/ewlsh/gjs-guide/-/blob/d4cb8a9fbdf669c791dc24bc1b052e404779157b/docs/extensions/upgrading/gnome-shell-45.md

**Comment by BigE** [comment]
Thanks for the report. I've been able to complete a bit of work recently and everything is in the develop branch that I have so far. I'll continue testing because I'm sure there are things I've missed and errors, but hopefully I can release soon to support 45!

**Comment by NastyaGrifon** [comment]
At [3db7a1b](https://github.com/BigE/desk-changer/commit/3db7a1b27efe7e0ed57f786514ad12f1e5ca9351) the extension is mostly working just fine with G45. Tested on Manjaro Gnome

Broken features atm:

- [ ] Show Icon as Preview shows empty image on toolbar
- [ ] Keyboard Shortcut Combination can't be changed and defaults to "Disabled"``

**Comment by BigE** [comment]
Thanks! Oddly enough, the preview bug only happens until the wallpaper is changed, then it shows up and works fine. That's been a source of pain (see #86) and needs a minor rewrite. As for preferences... they're getting a full rewrite to Adw currently, so that should resolve itself 🙃 Thanks again for helping to test things!

**Comment by belegdol** [comment]
It appears adding items to the Profile is not working either.

**Comment by BigE** [comment]
> It appears adding items to the Profile is not working either. I added a folder using dconf-editor for the time being.

Yes, I've got fixes for all of the bugs listed here, just haven't pushed them yet. I have some downtime this week, so I'll get everything synced up and hopefully make another release soon. Thanks!

**Comment by BigE** [comment]
Okay, preferences are rewritten and mostly working. I could have missed a thing or two, but you should be able to do all of the things from what's pushed into the [develop](https://github.com/BigE/desk-changer/tree/develop) branch. I'll work on getting this cleaned up a bit further and ready for a release... hopefully by the weekend.

**Comment by MacBadger** [comment]
cannot wait to get this working again!

**Comment by BigE** [comment]
Updated and released [version 36](https://github.com/BigE/desk-changer/releases/tag/version-36)
